### ISSUE TITLE
Improve plantuml errors

### ISSUE THREAD

**Issue by AlexanderLanin** [useblocks/sphinx-needs#754]
# Current state
When there is some problem in plantuml (e.g. via `needarch`) the only output we get is
```
WARNING: error while running plantuml

b'ERROR\n989\nSyntax Error?\nSome diagram description contains errors\n'
```

# Desired state
* What is the exact error? To my knowledge plantuml provides more info than what we see above! Either as text output or as generated image showing the error message.
* Was there a puml file generated I can look at? `:debug:` doesn't help here since I don't get the generated image. With some jinja and python magic I cannot tell what the plantuml code looks like which produced the error.

**Comment by danwos** [comment]
Thanks for the report :+1: 

Sphinx-Needs does not have direct access to  PlantUML itself.
It only generates the PlantUML code and gives it then to the used Sphinxcontrib-PlantUML library, which does all the rest.
Errors and other outputs are coming from this library, not from Sphinx-Needs.

However, the `:debug:`  flag is our workaround to give you at least something to reproduce the error in an environment, which is more under your control and where you get detailed error messages.

For this just copy the `:debug:` output to a file like ``test.puml`` and open it with the downloaded plantuml.jar, or copy it directly to the PlantUML online renderer: https://www.plantuml.com/plantuml/uml/

I'm not 100% sure, but afaik PlantUML renders its error messages also to an image, and this image gets embedded by the sphinxcontrib-plantuml lib into the rendered Sphinx HTML output as well. 
This may not be the case if ``-W`` is used for the build and the build stops at the first warning/error.

**Comment by AlexanderLanin** [comment]
> It only generates the PlantUML code and gives it then to the used Sphinxcontrib-PlantUML library

Therefore, it should be possible to retrieve the exact code given which led to an error.
Note: I cannot use `:save: my_needuml.puml` since I do not know which `needarch` invocation is responsible for the error.

> Errors and other outputs are coming from this library, not from Sphinx-Needs.

I'll try to write a ticket over there, thanks.

> This may not be the case if -W is used for the build and the build stops at the first warning/error.

Yeah, I don't get any image. Therefore `:debug:` is not helpful in this case.

**Comment by danwos** [comment]
> Therefore, it should be possible to retrieve the exact code given which led to an error.

I'm afraid it's not possible.
In details, Sphinx-Needs creates a `.. puml:: ` Directive inside the doctree and exit then.
Sphinx is then, at some time, reading the new `.. puml:: ` directive and executes it as it would be available in the rst code.

So there is no direct connection between the Plantuml-Extension and Sphinx-Needs, which makes it impossible in my eyes to get the error code.

**Comment by AlexanderLanin** [comment]
Part 1 ("What is the exact error?") solved via `plantuml_syntax_error_image = True` (https://github.com/sphinx-contrib/plantuml/)

**Comment by danwos** [comment]
Solved #982, but this includes also the PR https://github.com/sphinx-contrib/plantuml/pull/79#ref-pullrequest-1864983024. So the newest version of `sphinxcontrib-plantuml` must be used.
